cssparse/shared/src/main/scala/cssparse/CssParser.scala

```diff
@@ -0,0 +1,138 @@
+package cssparse
+
+import fastparse.all._
+
+// According to https://www.w3.org/TR/css-syntax-3/
+
+object CssParser {
+
+  import fastparse.all._
+  import fastparse.parsers.Terminals.AnyChar
+
+  val comment = P( "/*" ~/ (!"*/" ~ AnyChar).rep ~/ "*/")
+
+  val newline = P( "\n" | "\r\n" | "\r" | "\f")
+
+  val whitespace = P( " " | "\t" | newline)
+
+  val hex_digit = P( CharIn('0' to '9', 'a' to 'f', 'A' to 'F'))
+
+  val escape = P( "\\" ~ ((!(newline | hex_digit) ~ AnyChar) |
+    (hex_digit.rep(min=1, max=6) ~ whitespace.?)) )
+
+  val whitespace_token = P( comment | whitespace.rep(1) )
+
+  val ws = P( whitespace_token.rep )
+
+  val ident_token = P( ("-".? ~ (CharIn('a' to 'z', 'A' to 'Z', "_") | escape) ~
+    (CharIn('a' to 'z', 'A' to 'Z', '0' to '9', "_-") | escape).rep).! ) map Ast.IdentToken
+
+  val function_token = P( ident_token.! ~ "(" ) map Ast.FunctionToken
+
+  val at_word_token = P( "@" ~ ident_token.! ) map Ast.AtWordToken
+
+  val hash_token = P( "#" ~
+    (CharIn('a' to 'z', 'A' to 'Z', '0' to '9', "_-") | escape).rep(1).! ) map Ast.HashWordToken
```

Comment:
As a convention use the `.map(...)` syntax to be consistent with the rest of the code in the repo